src/intel/common/intel_check.c

```diff
@@ -93,3 +98,67 @@ bool intel_lower_ctx_priority(void)
    }
    return false;
 }
+
+bool has_local_mem(int fd)
+{
+   struct drm_i915_query_item item = {
+      .query_id = DRM_I915_QUERY_MEMORY_REGIONS,
+   };
+
+   struct drm_i915_query query = {
+      .num_items = 1, .items_ptr = (uintptr_t)&item,
+   };
+   if (drmIoctl(fd, DRM_IOCTL_I915_QUERY, &query)) {
+      mesa_loge("ioctl fail\n");
+      return false;
+   }
+   struct drm_i915_query_memory_regions *meminfo = calloc(1, item.length);
+   if (!meminfo) {
+      mesa_loge("memory fail\n");
+      return false;
+   }
+   item.data_ptr = (uintptr_t)meminfo;
+   if (drmIoctl(fd, DRM_IOCTL_I915_QUERY, &query) || item.length <= 0) {
+      free(meminfo);
+      mesa_loge("item.length <= 0\n");
+      return false;
+   }
+
+   uint64_t lmem_regions = 0;
+   for (uint32_t i = 0; i < meminfo->num_regions; i++) {
+      const struct drm_i915_memory_region_info *mem = &meminfo->regions[i];
+      if (mem->region.memory_class == I915_MEMORY_CLASS_DEVICE)
+         lmem_regions += 1;
+   }
+
+   free(meminfo);
+
+   return lmem_regions > 0;
+}
+
+bool is_dgpu(int fd)
+{
+   return has_local_mem(fd);
```

Comment:
has_local_mem is only called here, why need define one separate function for it?